vars/dockerBuildArgs.groovy

```diff
@@ -49,26 +101,43 @@ String call(Map config = [:]) {
       ret_str += " --build-arg CB0=" + current_time.get(Calendar.WEEK_OF_YEAR)
     }
 
-    if (add_repos) {
-      if (env.REPOSITORY_URL) {
-        ret_str += ' --build-arg REPO_URL=' + env.REPOSITORY_URL
-      }
-      if (env.DAOS_STACK_EL_7_LOCAL_REPO) {
-        ret_str += ' --build-arg REPO_EL7=' + env.DAOS_STACK_EL_7_LOCAL_REPO
+    String repo_base = env.REPOSITORY_URL
+    // No repo_base, no repos to add.
+    if (add_repos && repo_base) {
+      String repo_name = null
+      String repo_arg = ''
+      if (stage_info['target'] == 'centos7') {
+        repo_alias = 'EL_7'
+        repo_arg = 'EL7'
       }
-      if (env.DAOS_STACK_EL_8_LOCAL_REPO) {
-        ret_str += ' --build-arg REPO_EL8=' + env.DAOS_STACK_EL_8_LOCAL_REPO
+      if (stage_info['target'] == 'centos8') {
+        repo_alias = 'EL_8'
+        repo_arg = 'EL8'
       }
-      if (env.DAOS_STACK_LEAP_15_LOCAL_REPO) {
-        ret_str += ' --build-arg REPO_LOCAL_LEAP15=' +
-                   env.DAOS_STACK_LEAP_15_LOCAL_REPO
+      if (stage_info['target'] == 'leap15') {
+        repo_alias = 'LEAP_15'
+        repo_arg = 'LEAP'
```

Comment:
Should we have the version on this one, like we do for EL?

Reply:
Actually the tag just needs to match what is in the dockerfile.
For the leap dockerfile, I am refactoring down to one passed REPO from two, so can use a simpler repo_arg.

I need to double check that I did not break backwards compatibility in the last refactor though.

Reply:
Does the `repo_arg` need to even exist, or can we just use the same generic name in all of the Dockerfiles, since we are only passing the repos relevant to the distro?

Reply:
The current dockerfiles all use different names for the repo_arg parameter.  We would have to pass both for this change to be compatible with Daos, which has not yet been patched to request using one of the new repos.

The older code passed all the repos to all the Dockerfiles, and resulted in log noise for the build-args that were not used for the dockerfile.

We may want to phase in switching to a single --build-args parameter in a future PR.